stdlib/LinearAlgebra/src/triangular.jl

```diff
@@ -2607,7 +2607,9 @@ function sqrt(A::UpperTriangular{T},::Val{realmatrix}) where {T,realmatrix}
             @simd for k = i+1:j-1
                 r -= R[i,k]*R[k,j]
             end
-            iszero(r) || (R[i,j] = sylvester(R[i,i],R[j,j],-r))
+            if !(iszero(R[i,i]) && iszero(R[j,j]))
+                iszero(r) || (R[i,j] = sylvester(R[i,i],R[j,j],-r))
```

Comment:
Would be clearer to combine all of the `if` statements:
```jl
if !(iszero(r) || (iszero(R[i,i]) && iszero(R[j,j])))
    R[i,j] = sylvester(R[i,i],R[j,j],-r)
end
```

Reply:
changed to this in faa95fd